src/tests/ftest/pool/bad_connect.py

```diff
@@ -68,7 +68,7 @@ def test_connect(self):
             if expected_result in ['FAIL']:
                 self.fail("Test was expected to fail but it passed.\n")
 
-        except TestFail as excep:
+        except (TestFail, TypeError) as excep:
```

Comment:
@daltonbohning Thank you for your comment.
The original issue of the weekly failures (8 negative testcases) were due to bad pool.connect() with a Null uuid:
      **TypeError: an integer is required (got type NoneType)**

Before we were able to pass "Null" as uuid for the negative testcases, now the pydaos.api is just looking for an integer for uuid.
If we do not add the exception with the "TypeError", we have to use "0" as uuid for the negative testcases.
please comment.

Reply:
@dinghwah The problem is that this function:
https://github.com/daos-stack/daos/blob/136e9d9075bd42fbccfc4fbbe445320a59715052/src/tests/ftest/util/test_utils_pool.py#L143-L153
Calls this function:
https://github.com/daos-stack/daos/blob/0756f0f10e7de4190cba2c9cfcbd4094122bccd1/src/client/pydaos/raw/daos_api.py#L54-L56
Which calls this function:
https://github.com/daos-stack/daos/blob/0756f0f10e7de4190cba2c9cfcbd4094122bccd1/src/client/pydaos/raw/conversion.py#L10-L17

And none of those functions first check that `uuid is not None`. Ideally, I think the pydaos API (`c_uuid_to_str`) should have error checking, but I'm sure what our standard is in pydaos. Fixing this at the higher level works as well, since I think python `None` is translated to C `nullptr`.
```diff
diff --git a/src/tests/ftest/util/test_utils_pool.py b/src/tests/ftest/util/test_utils_pool.py
index dafc65e..306a651 100644
--- a/src/tests/ftest/util/test_utils_pool.py
+++ b/src/tests/ftest/util/test_utils_pool.py
@@ -148,7 +148,7 @@ class TestPool(TestDaosApiBase):

         """
         uuid = None
-        if self.pool:
+        if self.pool and self.pool.uuid:
             uuid = self.pool.get_uuid_str()
         return uuid
```

Reply:
Thanks for the great comments.
Original design for the negative tests were passing "Null" obj to test pool.connect() and expecting return failure, by updating the test_util.pool.py will satisfy the original test design.
Please see the new commit.

Reply:
No problem - thanks for updating!